src/core/io/configFB/io_controller_split.cpp

```diff
@@ -30,18 +30,13 @@ IOConfigFBSplitController::IOConfigFBSplitController(const TForteUInt8 *const pa
     scmSplitAdapterNum(paSplitAdapterNum),
     mSplitIterator(0) {
   mId = mInstancesIncrement++;
-  mInstances.pushBack(this);
+  mInstances.push_back(this);
 }
 
 IOConfigFBSplitController *IOConfigFBSplitController::getControllerById(TForteUInt16 paId) {
-  TControllerList::Iterator itEnd = mInstances.end();
-  int i = 0;
-  for (TControllerList::Iterator it = mInstances.begin(); it != itEnd; ++it, i++) {
-    if (paId == i && *it != nullptr) {
-      return *it;
-    }
-  }
-  return nullptr;
+  auto it = std::find_if(mInstances.begin(), mInstances.end(),
+                         [&](const IOConfigFBSplitController *obj) { return obj->mId == paId; });
```

Comment:
you just need `paId` inside the lambda, so no need to catch all by reference.

Reply:
Thx for pointing that out. I'm using these kind of clean-up for brushing up my C++ and missed it. Fixed now